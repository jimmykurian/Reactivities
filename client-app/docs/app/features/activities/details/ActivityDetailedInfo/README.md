[**Client App Documentation v0.0.0**](../../../../../README.md) â€¢ **Docs**

***

[Client App Documentation v0.0.0](../../../../../README.md) / app/features/activities/details/ActivityDetailedInfo

# app/features/activities/details/ActivityDetailedInfo

## Index

### Interfaces

- [Props](interfaces/Props.md)

### Functions

- [default](functions/default.md)
