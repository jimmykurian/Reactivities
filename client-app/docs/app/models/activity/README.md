[**Client App Documentation v0.0.0**](../../../README.md) â€¢ **Docs**

***

[Client App Documentation v0.0.0](../../../README.md) / app/models/activity

# app/models/activity

## Index

### Interfaces

- [Activity](interfaces/Activity.md)
