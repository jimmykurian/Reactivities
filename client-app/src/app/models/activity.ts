/**
 * @author Jimmy Kurian
 * @name Activity
 */

/**
 * The Activity interface represents the structure of an activity object in the Reactivities application.
 * It includes various properties such as id, title, date, description, category, city, and venue.
 *
 * @interface
 *
 * @property {string} id - The unique identifier for the activity.
 * @property {string} title - The title of the activity.
 * @property {string} date - The date of the activity in ISO string format.
 * @property {string} description - A brief description of the activity.
 * @property {string} category - The category to which the activity belongs.
 * @property {string} city - The city where the activity takes place.
 * @property {string} venue - The specific venue of the activity within the city.
 */
export interface Activity {
  id: string;
  title: string;
  date: string;
  description: string;
  category: string;
  city: string;
  venue: string;
}
