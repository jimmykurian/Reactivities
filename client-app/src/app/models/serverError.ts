/**
 * @file ServerError.ts
 * @author Jimmy Kurian
 * @fileoverview This file defines the `ServerError` interface, which models the structure of server error responses in the Reactivities application.
 */

/**
 * The ServerError interface represents the structure of an error returned from the server.
 * It includes the HTTP status code, an error message, and optional detailed information about the error.
 *
 * @interface ServerError
 *
 * @property {number} statusCode - The HTTP status code associated with the error.
 * @property {string} message - A brief message describing the error.
 * @property {string} details - Additional details or stack trace information about the error (optional).
 */
export interface ServerError {
  statusCode: number;
  message: string;
  details: string;
}
