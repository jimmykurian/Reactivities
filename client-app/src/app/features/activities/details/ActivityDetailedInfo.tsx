/**
 * @file ActivityDetailedInfo.tsx
 * @fileoverview This file contains the ActivityDetailedInfo component.
 * @author Jimmy Kurian
 */

import { observer } from 'mobx-react-lite';
import { Segment, Grid, Icon } from 'semantic-ui-react';
import { Activity } from '../../../models/activity';

/**
 * @interface Props
 * @description The properties for the ActivityDetailedInfo component.
 * @property {Activity} activity - The activity object containing details to display.
 */
export interface Props {
  activity: Activity;
}

/**
 * @component ActivityDetailedInfo
 * @description The ActivityDetailedInfo component displays detailed information about an activity.
 *
 * @param {Props} props - The properties object containing the activity to display.
 * @param {Activity} props.activity - The activity object containing details to display.
 *
 * @returns {JSX.Element} The rendered ActivityDetailedInfo component.
 *
 * @remarks
 * This component uses Semantic UI React components to create a segmented layout displaying the description, date, and venue of the activity.
 * Each segment includes an icon and the corresponding information from the activity object.
 *
 * @example
 * Here is an example of how to use the ActivityDetailedInfo component:
 * ```tsx
 * import React from 'react';
 * import ActivityDetailedInfo from './features/activities/details/ActivityDetailedInfo';
 * import { Activity } from '../../../models/activity';
 *
 * const activity: Activity = {
 *   id: '1',
 *   title: 'Sample Activity',
 *   description: 'This is a sample activity description.',
 *   date: '2024-01-01',
 *   city: 'Sample City',
 *   venue: 'Sample Venue',
 *   category: 'Sample Category',
 * };
 *
 * const App = () => (
 *   <div>
 *     <ActivityDetailedInfo activity={activity} />
 *   </div>
 * );
 *
 * export default App;
 * ```
 */
export default observer(function ActivityDetailedInfo({
  activity,
}: Props): JSX.Element {
  return (
    <Segment.Group>
      <Segment attached="top">
        <Grid>
          <Grid.Column width={1}>
            <Icon size="large" color="teal" name="info" />
          </Grid.Column>
          <Grid.Column width={15}>
            <p>{activity.description}</p>
          </Grid.Column>
        </Grid>
      </Segment>
      <Segment attached>
        <Grid verticalAlign="middle">
          <Grid.Column width={1}>
            <Icon name="calendar" size="large" color="teal" />
          </Grid.Column>
          <Grid.Column width={15}>
            <span>{activity.date}</span>
          </Grid.Column>
        </Grid>
      </Segment>
      <Segment attached>
        <Grid verticalAlign="middle">
          <Grid.Column width={1}>
            <Icon name="marker" size="large" color="teal" />
          </Grid.Column>
          <Grid.Column width={11}>
            <span>
              {activity.venue}, {activity.city}
            </span>
          </Grid.Column>
        </Grid>
      </Segment>
    </Segment.Group>
  );
});
