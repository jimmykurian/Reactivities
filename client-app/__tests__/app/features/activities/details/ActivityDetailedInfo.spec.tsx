import { render, screen } from '@testing-library/react';
import ActivityDetailedInfo, {
  Props,
} from '../../../../../src/app/features/activities/details/ActivityDetailedInfo';
import { faker } from '@faker-js/faker';
import { Activity } from '../../../../../src/app/models/activity';

// Mocking observer from mobx-react-lite
jest.mock('mobx-react-lite', () => ({
  observer: (component: React.ComponentType) => component,
}));

describe('ActivityDetailedInfo', () => {
  // Static mock activity for consistent snapshot tests
  const staticMockActivity: Activity = {
    id: '1',
    title: 'Static Title',
    description: 'This is a static description for testing.',
    date: '2024-06-12T00:00:00.000Z',
    city: 'Static City',
    venue: 'Static Venue',
    category: 'static-category',
  };

  const generateMockActivity = (): Activity => ({
    id: faker.string.uuid(),
    title: faker.lorem.words(3),
    description: faker.lorem.sentence(),
    date: faker.date.future().toISOString(),
    city: faker.location.city(),
    venue: faker.location.streetAddress(),
    category: faker.lorem.word(),
  });

  const renderComponent = (props: Props) => {
    render(<ActivityDetailedInfo {...props} />);
  };

  test('renders ActivityDetailedInfo component', () => {
    // Arrange
    const mockActivity = generateMockActivity();

    // Act
    renderComponent({ activity: mockActivity });

    // Assert
    expect(screen.getByText(mockActivity.description)).toBeInTheDocument();
    expect(screen.getByText(mockActivity.date)).toBeInTheDocument();
    expect(
      screen.getByText(`${mockActivity.venue}, ${mockActivity.city}`),
    ).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    // Arrange & Act
    const { asFragment } = render(
      <ActivityDetailedInfo activity={staticMockActivity} />,
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });
});
