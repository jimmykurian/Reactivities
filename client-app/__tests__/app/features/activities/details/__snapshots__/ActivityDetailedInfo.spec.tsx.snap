// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityDetailedInfo matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui segments"
  >
    <div
      class="ui top attached segment"
    >
      <div
        class="ui grid"
      >
        <div
          class="one wide column"
        >
          <i
            aria-hidden="true"
            class="teal info large icon"
          />
        </div>
        <div
          class="fifteen wide column"
        >
          <p>
            This is a static description for testing.
          </p>
        </div>
      </div>
    </div>
    <div
      class="ui attached segment"
    >
      <div
        class="ui middle aligned grid"
      >
        <div
          class="one wide column"
        >
          <i
            aria-hidden="true"
            class="teal calendar large icon"
          />
        </div>
        <div
          class="fifteen wide column"
        >
          <span>
            2024-06-12T00:00:00.000Z
          </span>
        </div>
      </div>
    </div>
    <div
      class="ui attached segment"
    >
      <div
        class="ui middle aligned grid"
      >
        <div
          class="one wide column"
        >
          <i
            aria-hidden="true"
            class="teal marker large icon"
          />
        </div>
        <div
          class="eleven wide column"
        >
          <span>
            Static Venue, Static City
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
